### ISSUE TITLE
Feature Request: Support multiple sync_dir/drive_id via one confdir

### ISSUE THREAD

**Issue by vvavrychuk** [abraunegg/onedrive#2310]
### Is your feature request related to a problem? Please describe.

I have to sync multiple SharePoint libraries, so for now I have to create different onedrive conf directories. But, this means that I have to authentificate onedrive multiple times for each config directory.

### Describe the solution you'd like

Maybe onedrive could parse multiple `*.config` files in single config directory.

### Describe alternatives you've considered

Alternative could be single config file with some hierarchy, e.g. YAML.

**Comment by abraunegg** [comment]
@vvavrychuk 
> I have to sync multiple SharePoint libraries, so for now I have to create different onedrive conf directories. 

Correct as this is the requirement of the application as per https://github.com/abraunegg/onedrive/blob/master/docs/SharePoint-Shared-Libraries.md

> But, this means that I have to authentificate onedrive multiple times for each config directory.

Not correct. You can symbolically link from 1 'master' configuration your authentication token (`refresh_token`) to each of your secondary configuration directories - with the caveat that they are all accessible from the same OneDrive Account.

> Maybe onedrive could parse multiple *.config files in single config directory.

This is not possible due to how the OneDrive API works - sorry.

**Comment by vvavrychuk** [comment]
> Not correct. You can symbolically link from 1 'master' configuration your authentication token (refresh_token) to each of your secondary configuration directories - with the caveat that they are all accessible from the same OneDrive Account.

But that means additional actions from my side. In addition I would like to sync all my SharePoint shared libraries using one `onedrive --synchronize` command. But now I have to run multiple synchronize commands for each SharePoint library.

I understand that it is not a critical issue, but maybe it make sense to do it for convinience.

>>Maybe onedrive could parse multiple *.config files in single config directory.
>
>This is not possible due to how the OneDrive API works - sorry.

Really, does onedrive config files are passed to OneDrive API as is? I don't understand limitation. I though onedrive config files are completely internal. So we can loop through all sync_dir/drive_id pairs in onedrive application.

**Comment by abraunegg** [comment]
@vvavrychuk 
> But that means additional actions from my side. In addition I would like to sync all my SharePoint shared libraries using one `onedrive --synchronize` command. But now I have to run multiple synchronize commands for each SharePoint library.

So why not:
* Create a shell script that calls all your SharePoint configurations in parallel so you are executing one command?
* Configure systemd services for each of your SharePoint sites to that they are syncing automatically in the background?

> Really, does onedrive config files are passed to OneDrive API as is? I don't understand limitation. I though onedrive config files are completely internal. So we can loop through all sync_dir/drive_id pairs in onedrive application.

Supporting your request would have the following impacts:
* Completely rework how the application is actually running - from forking to running multiple instances at the same time internally
* The backend database would have to be re-written to allow duplicate entries for a number of items such as the rootid and associated drive_id key pairs
* Support for all the client side filtering rules (skip_file, skip_dir, sync_list etc) would now need to be re-written so that they can be applied either at a global level or at an individual 'drive level' as your next ask would be why is 'skip_dir' being applied to all of my SharePoint drives as I only want it to be applied to X
* Threading and Forking & multi downloads - OneDrive API has a hard limitation of 5 active downloads - so if you have 10 SharePoint Sites you have in your 'bundled configuration' it is going to take infinitely longer to check and sync each site.

Whilst doing this may seem practical to you - it is not practical to how the OneDrive API operates. You will get better 'sync time efficiency' with having multi configurations and calling them from a shell script or enabling systemd to to sync them.

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
